File: graphs/Vertex.java
package graphs;

import java.util.Map;
import java.util.HashMap;
import java.util.Set;
import java.util.List;
import java.util.ArrayList;

public class Vertex {

    String key;
    Map<String,Integer> edges = new HashMap(); 
    List<Vertex> incomingEdges = new ArrayList(); 
    int totalEdges = 0;
    int distance;
    String color = "white";
    Vertex predecessor;
    

    public Vertex(String key) {
        this.key = key;
    }
    public String getKey() {
	return this.key;
    }
    public void addNeighbor(String key, int weight) {
	this.edges.put(key,weight);
	totalEdges++;
    }
    public Set<String> getNeighbors() {
	return this.edges.keySet();  //returns set of Keys
    }
    public Integer getCost(String neighbor) {
	return this.edges.get(neighbor);
    }
    public int getDistance() {
	return this.distance;
    }
    public String getColor() {
	return this.color;
    }
    public Vertex getPredecessor() {
	return this.predecessor;
    }
    public void setDistance(int distance) {
	this.distance = distance;
    }
    public void setColor(String color) {
	this.color = color;
    }
    public void setPredecessor(Vertex pred) {
	this.predecessor = pred;
    }
    public void addIncomingEdge(Vertex v) {
	this.incomingEdges.add(v);
    }
    public void removeIncomingEdge(Vertex v) {
	this.incomingEdges.remove(v);
    }
    public List<Vertex> getIncomingEdges() {
	return this.incomingEdges;
    }

}

File: graphs/python/Vertex.py
class Vertex(object):
    def __init__(self, key):
        self.id = key
        self.neighbors = {}

    def add_neighbor(self, neighbor, weight=0):
        self.neighbors[neighbor] = weight

    def get_neighbors(self):
        return self.neighbors

    def get_id(self):
        return self.id

    def get_weight(self, neighbor):
        return self.neighbors[neighbor]

    def __str__(self):
        return str(self.id) + ' neighbors: ' + str([x.id for x in self.neighbors])


if __name__ == "__main__":
    v1 = Vertex("A")
    v2 = Vertex("B")
    v3 = Vertex("C")
    v1.add_neighbor(v2,5)
    v1.add_neighbor(v3,10)
    v2.add_neighbor(v1,2)

    print v1
    print v2

    assert v1.get_weight(v2) == 5
    assert v1.get_weight(v3) == 10
    assert v2.get_weight(v1) == 2
    assert v1.get_id() == "A"
    assert v1.get_neighbors()[v2] == 5
    assert v1.get_neighbors()[v3] == 10
    

    
